context/app/static/js/pages/Organs/hooks.js

```diff
@@ -0,0 +1,49 @@
+import useSearchData from 'js/hooks/useSearchData';
+
+const query = {
+  size: 0,
+  query: {
+    bool: {
+      must: { term: { 'entity_type.keyword': 'Dataset' } },
+    },
+  },
+  aggs: {
+    'origin_sample.mapped_organ': { terms: { field: 'origin_sample.mapped_organ.keyword', size: 10000 } },
+  },
+};
+
+function buildOrganToCountMap(aggsBuckets) {
+  return aggsBuckets.reduce((acc, { key, doc_count }) => {
+    acc[key] = doc_count;
+    return acc;
+  }, {});
+}
+
+function addSearchTermsCount(search, organToCountMap) {
+  return search.reduce((sum, searchTerm) => sum + organToCountMap[searchTerm], 0);
+}
+
+function addDatasetCountsToOrgans(organs, aggsBuckets) {
+  const organToCountMap = buildOrganToCountMap(aggsBuckets);
+  return Object.entries(organs).reduce((acc, [k, { search }]) => {
+    acc[k] = { ...acc[k], descendantCounts: { Dataset: addSearchTermsCount(search, organToCountMap) } };
+    return acc;
+  }, organs);
+}
+
+function useOrgansDatasetCounts(organs) {
+  const { searchData, isLoading } = useSearchData(query);
+
+  if (isLoading) {
+    return { isLoading };
+  }
```

Comment:
Could you drop this? Or does something go wrong with `addDatasetCountsToOrgans`. Just trying to minimize branches...

Reply:
`addDatasetCountsToOrgans` errors without valid args, but I could change it to handle an empty array for `aggsBuckets` if preferable.